examples/ConfigurationStoreBackup/BackupAppConfigurationStore.cs

```diff
@@ -0,0 +1,196 @@
+    9. Delete all messages that we received in this batch from the storage queue.
+    10. Repeat from step 4 if there are any messages remaining in storage queue.
+ *
+ *
+ */
+
+using Azure;
+using Azure.Data.AppConfiguration;
+using Azure.Identity;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConfigurationStoreBackup
+{
+    public static class BackupAppConfigurationStore
+    {
+        private const string PrimaryConfigStoreEndpointEnvVarName = "PrimaryStoreEndpoint"; // eg., https://{store1}.azconfig.io
+        private const string SecondaryConfigStoreEndpointEnvVarName = "SecondaryStoreEndpoint"; // eg., https://{store2}.azconfig.io
+        private const string StorageQueueUriEnvVarName = "StorageQueueUri"; // eg., https://{account_name}.queue.core.windows.net/{queue_name}
+        private const int MaxMessagesToRead = 32;
+
+        [FunctionName("BackupAppConfigurationStore")]
+        public static async Task Run([TimerTrigger("0 */10 * * * *")]TimerInfo myTimer, ILogger log)
+        {
+            log.LogInformation($"Azure App Configuration store backup started at: {DateTime.Now}");
+            string storageQueueUri = Environment.GetEnvironmentVariable(StorageQueueUriEnvVarName);
+            if (string.IsNullOrEmpty(storageQueueUri))
+            {
+                throw new ArgumentException($"Please ensure that the environment variable '{StorageQueueUriEnvVarName}' is set to the storage queue URI.");
+            }
+
+            string primaryStoreEndpoint = Environment.GetEnvironmentVariable(PrimaryConfigStoreEndpointEnvVarName);
+            if (string.IsNullOrEmpty(primaryStoreEndpoint))
+            {
+                throw new ArgumentException($"Please ensure that the environment variable '{PrimaryConfigStoreEndpointEnvVarName}' is set to the endpoint of the primary App Configuration store.");
+            }
+
+            string secondaryStoreEndpoint = Environment.GetEnvironmentVariable(SecondaryConfigStoreEndpointEnvVarName);
+            if (string.IsNullOrEmpty(secondaryStoreEndpoint))
+            {
+                throw new ArgumentException($"Please ensure that the environment variable '{SecondaryConfigStoreEndpointEnvVarName}' is set to the endpoint of the secondary App Configuration store.");
+            }
+
+            QueueClient queueClient = new QueueClient(new Uri(storageQueueUri), new ManagedIdentityCredential());
+            ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(primaryStoreEndpoint), new ManagedIdentityCredential());
+            ConfigurationClient secondaryAppConfigClient = new ConfigurationClient(new Uri(secondaryStoreEndpoint), new ManagedIdentityCredential());
```

Comment:
Just realized we didn't have to create instantiate *AppConfigClient here. We only need to create them when there are actually messages in the queue. Can you please move them to `BackupKeyValuesAsync`?

Reply:
Okay, moved client instantiations to BackupAppConfigurationStoreAsync. Also moved QueueClient because it creates a logical grouping of all client instantiations in one method without any side effects.